extension/EdFi.Ods.Extensions.EdFiLMS/Artifacts/MsSql/Structure/Ods/1020-lms-user-google-classroom.sql

```diff
@@ -0,0 +1,60 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+CREATE OR ALTER PROCEDURE lms.harmonize_lmsuser_google_classroom AS
+BEGIN
+    SET NOCOUNT ON;
+
+-- Update based on any EdFi Student IdentificationCode matching with a LMS SISUserIdentifier
+    UPDATE 
+        lms.LMSUser
+    SET
+        EdFiStudentId = selectcodes.Id
+    FROM
+        (
+            SELECT
+                codes.IdentificationCode, s.Id
+            FROM
+                edfi.Student s
+            OUTER APPLY (
+                SELECT
+                    IdentificationCode  
+                FROM
+                    edfi.StudentEducationOrganizationAssociationStudentIdentificationCode sic
+                WHERE
+                    s.StudentUsi = sic.StudentUsi
+            ) AS codes
+        ) AS selectcodes
+    WHERE
+        SISUserIdentifier = selectcodes.IdentificationCode
+    AND
+        EdFiStudentId is NULL;
+        
+-- Update based on any EdFi Student Electronic Mail matching with a LMS SISUserIdentifier
+    UPDATE 
+        lms.LMSUser
+    SET
+        EdFiStudentId = selectemails.Id
+    FROM
+        (
+            SELECT
+                emails.ElectronicMailAddress, s.Id
+            FROM
+                edfi.Student s
+            OUTER APPLY (
+                SELECT
+                    ElectronicMailAddress  
+                FROM
+                    edfi.StudentEducationOrganizationAssociationElectronicMail sem
+                WHERE
+                    s.StudentUsi = sem.StudentUsi
+            ) AS emails
+        ) AS selectemails
+    WHERE
+        SISUserIdentifier = selectemails.ElectronicMailAddress
+    AND
+        EdFiStudentId is NULL;
```

Comment:
Do we want to rollback the first update if the second one fails - and thus need to open a transaction?

I think the answer is no - we are not losing anything by keeping the initial change in place. Its not like we destroyed some existing data.

So I'm happy with the current state of affairs.